src/test/java/codeu/model/store/basic/IndexStoreTest.java

```diff
@@ -70,4 +100,35 @@ public void testSplitAndHashMessage () {
     Assert.assertEquals(false, wordsHash.containsKey("one."));
   }
 
+  @Test
+  public void testsortMessageList () {
+    Assert.assertEquals(inOrderMessageList, indexStore.search("message"));
+  }
+
+  @Test
+  public void testSearchIntersection () {
+    List<Message> expectedList = new ArrayList<>();
+    expectedList.add(MESSAGE_ONE);
+    expectedList.add(MESSAGE_FOUR);
+    Assert.assertEquals(expectedList, indexStore.search("test&&one&&message"));
+    Assert.assertEquals(expectedList, indexStore.search("test&&one&&"));
+    Assert.assertEquals(expectedList, indexStore.search("&&one&&test"));
+
+  }
+
+  @Test
+  public void testSearchUnion () {
+    List<Message> expectedList = new ArrayList<>();
+    expectedList.add(MESSAGE_TWO);
```

Comment:
Add union and intersection tests for case when the expectedList is empty or null